rts/motoko-rts/src/stream.rs

```diff
@@ -14,13 +14,15 @@
 
 // Layout of a stream node:
 //
-//      ┌────────────┬─────┬───────┬─────────┬─────────┬───────────┬────────┬──────────┐
-//      │ tag (blob) │ len │ ptr64 │ start64 │ limit64 │ outputter │ filled │ cache... │
-//      └────────────┴─────┴───┴───┴────┴────┴────┴────┴───────────┴────────┴──────────┘
+// ┌────────────┬─────┬─────────┬───────┬─────────┬─────────┬───────────┬────────┬──────────┐
+// │ obj header │ len │ padding | ptr64 │ start64 │ limit64 │ outputter │ filled │ cache... │
+// └────────────┴─────┴─────────┴───────┴─────────┴─────────┴───────────┴────────┴──────────┘
 //
 // We reuse the opaque nature of blobs (to Motoko) and stick Rust-related information
 // into the leading bytes:
-// - `tag` and `len` are blob metadata
+// - `obj header` contains tag and forwarding pointer
+// - `len` is in blob metadata
+// - 'padding' to align to 64-bit
```

Comment:
Actually, it is critical, because the Rust compiler (according to C-memory representation) automatically aligns the next field `ptr64` to 64-bit. I only added this field for clarity. I took me a while to figure out that bug...

Reply:
> I took me a while to figure out that bug...
I bet!